components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -204,7 +202,7 @@ class DownloadsFeature(
 
         withActiveDownload { (tab, download) ->
             if (applicationContext.isPermissionGranted(downloadManager.permissions.asIterable())) {
-                processDownload(tab, download)
+                startDownload(download)
```

Comment:
This is a cleaver solution but it's causing another issue, because when we receive a download request without the external download manager option active, the first time the permissions dialog is going to show, if we allow the permissions, the download is going to start right away without showing the confirmation dialog.

Reply:
Sorry for all the back and forth, I just want to make sure we add the need feature without breaking the existing one, I really appreciate all the effort that you have put on the feature ❤️ !

Reply:
> 
> 
> This is a cleaver solution but it's causing another issue, because when we receive a download request without the external download manager option active, the first time the permissions dialog is going to show, if we allow the permissions, the download is going to start right away without showing the confirmation dialog.

so i should surround this with if statement? If external download manager active -> startDownload, else processDownload?

Reply:
> 
> 
> Sorry for all the back and forth, I just want to make sure we add the need feature without breaking the existing one, I really appreciate all the effort that you have put on the feature ❤️ !

Thank you, I'm happy to help you guys. I'm a student and i don't have money to donate (and some weird exchange rate in my country)(also i heard that money doesn't went to development) , soo i'm doing what i can :)

Reply:
> so i should surround this with if statement? If external download manager active -> startDownload, else processDownload?


That sounds good, also after `startDownload` call `useCases.consumeDownload(tab.id, download.id)` as `startDownload` doesn't consume the download, we will need to update the tests for this new behavior.

Reply:
Thanks to you I noticed there is an bug in downloads, it was there for a long time https://github.com/mozilla-mobile/fenix/issues/17650 :)

Reply:
@Amejia481 tests currently passing but not testing this behavior and i'm unable to test new behavior :(

Reply:
🎉